### PR TITLE
Create user model Django

### PR DESCRIPTION
**Pull request by Samox**

### CODE REVIEW

**Comment by bam-bot-tech** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  **<a href="https://github.com/bamlab/dev-standards/blob/create-user-model-django/backend/django/create-user-model.mo.md">create-user-model.mo.md</a>**: Seems you do not need a *Troubleshoot* part
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by tychota** [comment]
@Samox: thank you for writing this.

You can use:
- https://github.com/bamlab/dev-standards/blob/master/backend/graphql-js/getting-started-with-apollo-server-dataloader-knex.mo.md
- https://github.com/bamlab/dev-standards/blob/master/backend/django/getting-started.mo.md

as example of good standards.

Call me if you need any help to act your Andon.

File: backend/django/create-user-model.mo.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Coroute lol
```
Comment (tychota) [inline backend/django/create-user-model.mo.md]: Tu as fais une typo dans `coroute` !

Comment (louiszawadzki) [inline backend/django/create-user-model.mo.md]: *Croute (typo)

Comment (josuelengronne) [reply backend/django/create-user-model.mo.md]: ~~You could put an emoji, be a bit more 2017!~~ What about no ! (edit Tycho)

**Review by josuelengronne — DISMISSED** [review]

File: backend/django/create-user-model.mo.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# Context:
+You have a Django project installed. You need to handle different types of users
```
Comment (tychota) [inline backend/django/create-user-model.mo.md]: I would like a more precise description of the example.

> You have a Django project installed. 
> You need to handle different a normal django user (you, as an admin), but also some `Mayor` that have extra fields.

File: backend/django/create-user-model.mo.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+# Prerequisite:
+Have the project installed.
```
Comment (tychota) [inline backend/django/create-user-model.mo.md]: Perfect, just make it a checklist and add the Prequesite time.

> # Prerequisite (30min)
> - have the project installed: see this [method of operation](https://github.com/bamlab/dev-standards/blob/master/backend/django/getting-started.mo.md)

File: backend/django/create-user-model.mo.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+https://github.com/bamlab/dev-standards/blob/master/backend/django/getting-started.mo.md
+
+# Situation:
+You have a basic user model with a name.
+You want to create another type of user, a Mayor that has a City.
```
Comment (tychota) [inline backend/django/create-user-model.mo.md]: Insist on the fact that city is a foreign key to other model

File: backend/django/create-user-model.mo.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+# Prerequisite:
+Have the project installed.
+https://github.com/bamlab/dev-standards/blob/master/backend/django/getting-started.mo.md
+
+# Situation:
+You have a basic user model with a name.
+You want to create another type of user, a Mayor that has a City. 
+You do not want to override the classic user model.
+You want to create a Mayor from the admin
+You want the newly created Mayor to be part of the staff (User that can access the admin)
```
Comment (tychota) [inline backend/django/create-user-model.mo.md]: I would say:

> staff but not superuser.

the difference is not clear for beginners

File: backend/django/create-user-model.mo.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+## Add the Mayor model
```
Comment (tychota) [inline backend/django/create-user-model.mo.md]: How much times does it takes ?

File: backend/django/create-user-model.mo.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+- In the models.py file, where the User is defined, create a new class that inherits the User:
+```
+class Mayor(User):
+    city = models.ForeignKey('locations.city')
+
+    class Meta:
+        verbose_name = "Mayor"
+```
+
```
Comment (tychota) [inline backend/django/create-user-model.mo.md]: # Steps missing

- run the `makemigrations` command
- run the `migrate` command

# Checks missing

- run the `showmigrations` command

File: backend/django/create-user-model.mo.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    class Meta:
+        verbose_name = "Mayor"
+```
+ 
+## Create a Mayor from the admin
+
+- In the admin.py, where the User model is added `admin.register(User)`
+- Add the following:
+```
+@admin.register(Mayor)
```
Comment (tychota) [inline backend/django/create-user-model.mo.md]: - Add the import of Mayor in this file.
- Add the import of UserAdmin

File: backend/django/create-user-model.mo.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+    class Meta:
+        verbose_name = "Mayor"
+```
+ 
+## Create a Mayor from the admin
+
+- In the admin.py, where the User model is added `admin.register(User)`
+- Add the following:
+```
```
Comment (tychota) [inline backend/django/create-user-model.mo.md]: use ` ```python` to be correctly syntax highlited

File: backend/django/create-user-model.mo.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    pass
+```
+
+You now see the Mayor in the admin. But all the forms (form to add a Mayor, form to change/update a Mayor) are the same as for the regular User.
+
+## Override the admin form to add a Mayor
+
+- In the admin.py when you declared your MyMayorAdmin class, you can choose the fields you want to display in the form by overriding the "add_fieldsets" property.
+- In our case:
+```
```
Comment (tychota) [inline backend/django/create-user-model.mo.md]: same, python highlight

File: backend/django/create-user-model.mo.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+@admin.register(Mayor)
+class MyMayorAdmin(UserAdmin):
+    pass
+```
+
+You now see the Mayor in the admin. But all the forms (form to add a Mayor, form to change/update a Mayor) are the same as for the regular User.
+
+## Override the admin form to add a Mayor
+
+- In the admin.py when you declared your MyMayorAdmin class, you can choose the fields you want to display in the form by overriding the "add_fieldsets" property.
```
Comment (tychota) [inline backend/django/create-user-model.mo.md]: link to the django doc welcomed

Example:

> You can readme more about add_fieldset here: https://docs.djangoproject.com/en/1.11/topics/auth/customizing/#a-full-example

My goal is to show bamers that django doc is awesome and that they can find so followup there.

File: backend/django/create-user-model.mo.md (hunk 10 above)
Comment (tychota) [reply backend/django/create-user-model.mo.md]: And same for import. I would like the MO to be easily reusable and for that the reader should not search for the imports.

File: backend/django/create-user-model.mo.md [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+- Add the following:
+```
+@admin.register(Mayor)
+class MyMayorAdmin(UserAdmin):
+    pass
+```
+
+You now see the Mayor in the admin. But all the forms (form to add a Mayor, form to change/update a Mayor) are the same as for the regular User.
+
+## Override the admin form to add a Mayor
```
Comment (tychota) [inline backend/django/create-user-model.mo.md]: Time

File: backend/django/create-user-model.mo.md [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+- In the admin.py when you declared your MyMayorAdmin class, you can choose the fields you want to display in the form by overriding the "add_fieldsets" property.
+- In our case:
+```
+class MyMayorAdmin(AuthUserAdmin):
+    add_fieldsets = (
+            ('User Profile', {'fields': ('username', 'city', 'password1', 'password2')}),
+    )
+```
+
+## Give admin property to mayor user
```
Comment (tychota) [inline backend/django/create-user-model.mo.md]: Time

File: backend/django/create-user-model.mo.md [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+
+- You can manually set a user to be part of the staff, you can also override the save method of the AddForm so it changes the property is_staff to True
+- First create a form that inherits the AddUserForm:
```
Comment (tychota) [inline backend/django/create-user-model.mo.md]: You should say exactly what you do in the code, with link to doc

> First create a form that inherits the AddUserForm and override the `save` method (see [overriding custom model method](https://docs.djangoproject.com/en/1.11/topics/db/models/#overriding-predefined-model-methods))

File: backend/django/create-user-model.mo.md [hunk 15]
```diff
@@ -0,0 +1,74 @@
+# Context:
+You have a Django project installed. You need to handle different types of users
+
+# Prerequisite:
+Have the project installed.
+https://github.com/bamlab/dev-standards/blob/master/backend/django/getting-started.mo.md
+
+# Situation:
+You have a basic user model with a name.
```
Comment (tychota) [inline backend/django/create-user-model.mo.md]: Checklist please. It render to 
image

**Review by tychota — APPROVED** [review]
Some inline comments to be addressed.

The respect what danger said:

image

After this, it is validated and you can merge.

File: backend/django/create-user-model.mo.md (hunk 7 above)
Comment (Samox) [reply backend/django/create-user-model.mo.md]: trop stylé ca je connaissais pas

**Comment by tychota** [comment]
Typo on readme.md url.
Missing link in summary.md

**Pull request merged**